src/Controls/src/Core/Platform/Windows/VisualElementTracker.cs

```diff
@@ -555,12 +555,15 @@ void OnPointerCanceled(object sender, PointerRoutedEventArgs e)
 
 		void OnPointerExited(object sender, PointerRoutedEventArgs e)
 		{
-			uint id = e.Pointer.PointerId;
-			if (_fingers.Contains(id))
-				_fingers.Remove(id);
+			if (!_isPanning)
+			{
+				uint id = e.Pointer.PointerId;
+				if (_fingers.Contains(id))
+					_fingers.Remove(id);
+			}
+
 			SwipeComplete(true);
 			PinchComplete(true);
```

Comment:
Why are the swipe and pinched canceled here still? Should this all maybe be on pointer up?

Reply:
If you swipe or pinch the Completed status is sent if the pointer leaves the view. Are also setting the Complete state when the pointer up.